### PR TITLE
Json Serialization

### PR DESCRIPTION
**Pull request by nouda**
Clients can now choose the message serialization protocol.
The options are (i) Protobuf or (ii) JSON with Protobuf as default.

The new URL parameter / properties key "serialization" can be used to specify the protocol.
The value is not dependent on case, i.e., clients can specify, for example, "serialization=jSoN".

The former, not used, parameter key "wire_protocol" is deprecated and sets "serialization" if "serialization" is not present. (I.e., "wire_protocol" does not overwrite "serialization" if both are specified.)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -55,15 +54,11 @@ compileJava   {
     targetCompatibility = 1.8
 }
 
-configurations.all {
-    // check for updates every build
-    resolutionStrategy.cacheChangingModulesFor 0, "seconds"
-}
-
```
Comment (vogti) [inline build.gradle]: Why did you remove this?

Comment (nouda) [reply build.gradle]: Re-added in b8c9f819de34c2ae7220db45d3e469e15d0a61c3.
